scripts/deploy/iks/deploy-kfp.sh

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+#
+# Copyright 2021 kubeflow.org
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Remove the x if you need no print out of each command
+set -ex
+
+# Need the following env
+# - PIPELINE_KUBERNETES_CLUSTER_NAME:   kube cluster name
+# - KUBEFLOW_NS:                        namespace for kfp-tekton, defulat: kubeflow
+
+MAX_RETRIES="${MAX_RETRIES:-5}"
+SLEEP_TIME="${SLEEP_TIME:-10}"
+EXIT_CODE=0
+
+KUSTOMIZE_DIR="${KUSTOMIZE_DIR:-"manifests/kustomize/env/platform-agnostic"}"
+MANIFEST="${MANIFEST:-"kfp-tekton.yaml"}"
+KUBEFLOW_NS="${KUBEFLOW_NS:-kubeflow}"
+
+API_SERVER_IMAGE="${API_SERVER_IMAGE:-"docker.io/aipipeline/api-server"}"
+NEW_API_SERVER_IMAGE="${NEW_API_SERVER_IMAGE:-"${REGISTRY_URL}/${REGISTRY_NAMESPACE}/api-server:${IMAGE_TAG}"}"
+
+METADATA_WRITER_IMAGE="${METADATA_WRITER_IMAGE:-"docker.io/aipipeline/metadata-writer"}"
+NEW_METADATA_WRITER_IMAGE="${NEW_METADATA_WRITER_IMAGE:-"${REGISTRY_URL}/${REGISTRY_NAMESPACE}/metadata-writer:${IMAGE_TAG}"}"
+
+PERSISTENCEAGENT_IMAGE="${PERSISTENCEAGENT_IMAGE:-"docker.io/aipipeline/persistenceagent"}"
+NEW_PERSISTENCEAGENT_IMAGE="${NEW_PERSISTENCEAGENT_IMAGE:-"${REGISTRY_URL}/${REGISTRY_NAMESPACE}/persistenceagent:${IMAGE_TAG}"}"
+
+SCHEDULEDWORKFLOW_IMAGE="${SCHEDULEDWORKFLOW_IMAGE:-"docker.io/aipipeline/scheduledworkflow"}"
+NEW_SCHEDULEDWORKFLOW_IMAGE="${NEW_SCHEDULEDWORKFLOW_IMAGE:-"${REGISTRY_URL}/${REGISTRY_NAMESPACE}/scheduledworkflow:${IMAGE_TAG}"}"
+
+SA_PATCH=$(cat << EOF
```

Comment:
A short comment here about the patch might be helpful.

Reply:
good point and I also found I forgot to copy the `all-icr-io` secret into target namespace(kubeflow). fixed